platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/internal/cc/impl/ConfigurationCacheGracefulDegradationIntegrationTest.groovy

```diff
@@ -0,0 +1,244 @@
+            ${generateDegradationController()}
+            tasks.register("foo") { task ->
+                degradation.requireConfigurationCacheDegradation(task, provider { "Project access" })
+                doLast {
+                    println "Hello from foo \${project.path}"
+                }
+            }
+
+            tasks.register("bar") {
+                doLast {
+                    println "Hello from bar \${project.path}"
+                }
+            }
+        """
+
+        when:
+        configurationCacheFails "foo", "bar"
+
+        then:
+        configurationCache.assertNoConfigurationCache()
+        problems.assertFailureHtmlReportHasProblems(failure) {
+            totalProblemsCount = 2
+            withProblem("Build file 'build.gradle': line 11: invocation of 'Task.project' at execution time is unsupported.")
+            withProblem("Build file 'build.gradle': line 5: invocation of 'Task.project' at execution time is unsupported.")
+            withIncompatibleTask(":foo", "Project access.")
+        }
+    }
+
+    def "a task in included build can require CC degradation"() {
+        def configurationCache = newConfigurationCacheFixture()
+        buildFile("buildLogic/build.gradle", """
+            ${generateDegradationController()}
+            tasks.register("foo") { task ->
+                degradation.requireConfigurationCacheDegradation(task, provider { "Project access" })
+                doLast {
+                    println "Hello from included build \${project.path}"
+                }
+            }
+        """)
+        settingsFile """
+            includeBuild("buildLogic")
+        """
+
+        when:
+        configurationCacheRun ":buildLogic:foo"
+
+        then:
+        configurationCache.assertNoConfigurationCache()
+        problems.assertResultHtmlReportHasProblems(result) {
+            totalProblemsCount = 1
+            withProblem("Build file 'buildLogic/build.gradle': line 5: invocation of 'Task.project' at execution time is unsupported.")
+            withIncompatibleTask(":buildLogic:foo", "Project access.")
+        }
+
+        and:
+        outputContains("Hello from included build :")
+        postBuildOutputContains("""
+Configuration caching disabled because degradation was requested by:
+- task `:buildLogic:foo` of type `org.gradle.api.DefaultTask`""")
+    }
+
+    def "depending on a CC degrading task from included build introduces CC degradation"() {
+        def configurationCache = newConfigurationCacheFixture()
+        buildFile("buildLogic/build.gradle", """
+            ${generateDegradationController()}
+            tasks.register("foo") { task ->
+                degradation.requireConfigurationCacheDegradation(task, provider { "Project access" })
+                doLast {
+                    println "Hello from included build \${project.path}"
+                }
+            }
+        """)
+        settingsFile """
+            includeBuild("buildLogic")
```

Comment:
💅 The name "buildLogic" for these builds is confusing because they are not providing "build logic", i.e., plugins

Reply:
Fixed in the new PR